src/components/BannerTypes.ts

```diff
@@ -44,9 +45,11 @@ export type CanRun = (targeting: BannerTargeting, pageTracking: BannerPageTracki
 
 export interface BannerTest {
     name: string;
+    path: string;
+    bannerType: 'contributions' | 'subscriptions';
     canRun: CanRun;
     minPageViews: number;
-    variants: BannerVariant[];
+    variants: BannerVariant[] | [];
```

Comment:
why is the `| []` necessary?

Reply:
We could make variants optional, then we wouldn't have to pass an empty array if there aren't any

Reply:
Does typescript complain about setting this to an empty array if it's `BannerVariant[]`? That is surprising...

Reply:
It wasn't happy about it. I have changed variant to be optional, which has meant that in the interface `BannerTestSelection`, variant has to be `variant: BannerVariant | null;` but that seems like a better trade-off.